src/libraries/System.ComponentModel.Composition/src/System/ComponentModel/Composition/Hosting/CatalogExportProvider.cs

```diff
@@ -159,7 +157,7 @@ public ExportProvider SourceProvider
                 ThrowIfDisposed();
                 using (_lock.LockStateForRead())
                 {
-                    return _sourceProvider;
+                    return _sourceProvider!;  // Should be [DisallowNull]
```

Comment:
I think this property should be annotated as nullable with `[DisallowNull]` as setter throw on null but getter could return null before it set with a non null value. 

The problem here with `[DisallowNull]` attribute setter `value` not became non null, and warning